server/src/main/java/org/opensearch/search/SearchHit.java

```diff
@@ -559,15 +536,29 @@ public String getClusterAlias() {
         return clusterAlias;
     }
 
-    public void matchedQueries(String[] matchedQueries) {
+    public void matchedQueries(Map<String, Float> matchedQueries) {
```

Comment:
This is a breaking change of public API (see please this class annotations) for `2.x`, please keep old method as well.

Reply:
done